lib/insulet/insuletSimulator.js

```diff
@@ -346,6 +346,10 @@ exports.make = function(config){
       // (although within their own type all the basals are always in order)
       // end result: we have to sort events again before we try to upload them
       var orderedEvents = _.sortBy(_.filter(events, function(event) {
```

Comment:
I believe it's technically OK for filter to modify the elements it touches but it does feel a bit wrong to me. I think I'd at least make note of it in a comment.

Also, because you've nested sort and filter, it's hard to tell what's going on. Any chance of pulling filter out into a named local function?